File: 20.valid-parentheses.ts
/*
 * @lc app=leetcode id=20 lang=typescript
 *
 * [20] Valid Parentheses
 */

// @lc code=start
function isValid(s: string): boolean {
  const left = ["(", "{", "["];
  const right = [")", "}", "]"];
  let sArray: string[] = s.split("");
  let stack: string[] = [];
  for (let i = 0; i < sArray.length; i++) {
    if (left.includes(sArray[i])) {
      stack.push(sArray[i]);
    } else {
      if (right.indexOf(sArray[i]) === left.indexOf(stack[stack.length - 1])) {
        stack.pop();
      } else {
        return false;
      }
    }
  }
  if (stack.length !== 0) return false;
  return true;
}
// @lc code=end

File: 20.valid-parentheses.py
#
# @lc app=leetcode id=20 lang=python3
#
# [20] Valid Parentheses
#

# @lc code=start
class Solution:
    def isValid(self, s: str) -> bool:
        left_stack = []
        for i in range(len(s)):
            print(left_stack)
            if s[i] == '(' or s[i] == '[' or s[i] == '{':
                left_stack.append(s[i])
            else:
                if s[i] == ')':
                    if len(left_stack) == 0 or left_stack.pop() != '(':
                        return False
                if s[i] == ']':
                    if len(left_stack) == 0 or left_stack.pop() != '[':
                        return False
                if s[i] == '}':
                    if len(left_stack) == 0 or left_stack.pop() != '{':
                        return False
        if len(left_stack):
            return False
        return True
# @lc code=end
